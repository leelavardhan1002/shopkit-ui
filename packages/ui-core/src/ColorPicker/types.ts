export interface Color {
  id: string;
  label: string;
  value: string;
  borderColor?: string;
  selectedBorderColor?: string;
  disabled?: boolean;
}

export interface CoreProps {
  colors: Array<Color>;
  colorValue?: Color;
  defaultColorValue?: Color;
  onColorChange?: (color?: Color) => void;
  disabled?: boolean;
}

export interface AppearanceProps {
  variant?: 'square' | 'pill';
  size?: 'small' | 'medium' | 'large';
  showTooltip?: boolean;
  showLabels?: boolean;
  borderRadius?: number;
  maxVisible?: number;
  maxVisibleLabel?: string;
}

export interface ColorPickerProps extends CoreProps, AppearanceProps {
  className?: string;
  'aria-label'?: string;
}
