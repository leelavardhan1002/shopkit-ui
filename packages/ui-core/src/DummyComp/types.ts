export interface DummyCompProps {
    // Add your props here
}