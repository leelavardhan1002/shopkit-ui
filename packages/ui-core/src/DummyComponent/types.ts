export interface DummyComponentProps {
    // Add your props here
}