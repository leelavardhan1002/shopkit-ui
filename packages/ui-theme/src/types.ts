import { coreTokens } from './data/coreThemeTokens';

declare module '@mui/material/styles' {
    interface Theme {
        sys: typeof coreTokens.sys;
    }
}

export interface ShopkitSysThemeOptions {
    palette?: typeof coreTokens.palette;
    sys: typeof coreTokens.sys;
}

export interface ShopkitThemeProviderProps {
    children: React.ReactNode;
    themeOverrides?: ShopkitSysThemeOptions;
}
