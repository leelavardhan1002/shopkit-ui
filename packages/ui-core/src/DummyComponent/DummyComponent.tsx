import { DummyComponentProps } from './types';
import { StyledDummyComponent } from './DummyComponent.style';

export const DummyComponent: React.FC<DummyComponentProps> = (props) => {
    return (
        <StyledDummyComponent>
            DummyComponent
        </StyledDummyComponent>
    );
};