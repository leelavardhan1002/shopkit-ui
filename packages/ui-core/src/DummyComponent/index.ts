export { DummyComponent } from './DummyComponent';
export type { DummyComponentProps } from './types';