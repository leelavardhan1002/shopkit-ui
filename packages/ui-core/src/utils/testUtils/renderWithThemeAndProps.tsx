import { render } from "@testing-library/react";
import { ShopKitThemeProvider } from "@shopkit/ui-theme";
import React from "react";

export const renderWithTheme = (component: React.ReactElement) => {
    return render(
        <ShopKitThemeProvider>
            {component}
        </ShopKitThemeProvider>
    );
};
