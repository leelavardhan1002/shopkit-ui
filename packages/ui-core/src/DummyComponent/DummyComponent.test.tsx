import { DummyComponent } from './DummyComponent';
import { renderWithTheme } from '../utils/testUtils/renderWithThemeAndProps';
import { DummyComponentProps } from './types';

const defaultProps: Partial<DummyComponentProps> = {};

const setupElement = (props: DummyComponentProps) => {
    return renderWithTheme(<DummyComponent {...props} />);
}

describe('DummyComponent', () => {
    it('matches snapshot', () => {
        const { asFragment } = setupElement(defaultProps);
        expect(asFragment()).toMatchSnapshot();
    });
});
