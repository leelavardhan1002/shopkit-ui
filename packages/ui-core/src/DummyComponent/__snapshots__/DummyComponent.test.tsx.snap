// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-xviiwl"
  >
    DummyComponent
  </div>
</DocumentFragment>
`;
